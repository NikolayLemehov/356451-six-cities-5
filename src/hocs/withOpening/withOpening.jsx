import React, {PureComponent} from "react";

const withOpening = (Component) => {
  class WithOpening extends PureComponent {
    constructor(props) {
      super(props);
      this.state = {
        isOpened: false,
      };
      this.handleMenuClose = this.handleMenuClose.bind(this);
      this.handleMenuToggle = this.handleMenuToggle.bind(this);
    }

    handleMenuClose() {
      this.setState({isOpened: false});
    }

    handleMenuToggle(isOpened) {
      this.setState({isOpened: !isOpened});
    }

    render() {
      const {isOpened} = this.state;

      return (
        <Component
          {...this.props}
          isOpened={isOpened}
          onCloseMenu={this.handleMenuClose}
          onToggleMenu={this.handleMenuToggle}
        />
      );
    }
  }

  return WithOpening;
};

export default withOpening;
