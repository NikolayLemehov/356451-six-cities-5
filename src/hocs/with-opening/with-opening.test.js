import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withOpening from "./with-opening";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withOpening(MockComponent);

it(`withOpening is rendered correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped>
      <React.Fragment />
    </MockComponentWrapped>
  )).toJSON();

  expect(tree).toMatchSnapshot();
});
