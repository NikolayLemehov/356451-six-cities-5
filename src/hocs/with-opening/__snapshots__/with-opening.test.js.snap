// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withOpening is rendered correctly 1`] = `<div />`;
